velox/exec/Driver.h

```diff
@@ -244,6 +244,43 @@ struct DriverCtx {
   std::optional<common::SpillConfig> makeSpillConfig(int32_t operatorId) const;
 };
 
+/// Enum for operator methods we want to track.
+enum class OperatorMethod {
+  kNone,
+  kIsBlocked,
+  kNeedsInput,
+  kGetOutput,
+  kAddInput,
+  kNoMoreInput
+};
+
+/// Structure holds the information about the current operator call the driver
+/// is in. Can be used to detect deadlocks and otherwise blocked calls.
+/// If timeStartMs is zero, then we aren't in an operator call.
+struct OpCallStatus {
+  /// Time (ms) when the operator call started.
+  std::atomic_size_t timeStartMs{0};
+  /// Id of the operator, method of which is currently running. It is index into
+  /// the vector of Driver's operators.
+  std::atomic_int32_t opId{0};
+  /// Method of the operator, which is currently running.
+  std::atomic<OperatorMethod> method{OperatorMethod::kNone};
```

Comment:
Just `std::atomic<const char*>` should be enough (since all the strings are constant), then you don't need to do the dispatch yourself